packages/suite/src/reducers/suite/deviceReducer.ts

```diff
@@ -138,12 +138,18 @@ const changeDevice = (
     if (!device.features) return;
 
     // find devices with the same "device_id"
-    const affectedDevices = draft.filter(
-        d =>
+    const affectedDevices = draft.filter(d => {
+        const parsedDeviceState = d.state?.split(/@|:/); // address@device_id:instance
```

Comment:
not sure that this will be the best approach, device state format may change in the future.
we set current state/format for debugging purposes and never went back but i think it should be a shorter hash (see[ this commented line](https://github.com/trezor/connect/blob/develop/src/js/device/DeviceCommands.js#L323))

 maybe we should always prevent device_id from overrinding when switched to bootloader? (see merge func. above)  or maybe we should introduce additional field in device for that purpose?

Reply:
I preserved id if it wanted to be changed to null. c5b217a48d721263d0e398cfd8c3537dc647c778

Reply:
could you pls double check if there are places where device.id is **expected** to be null?

Reply:
Double checked, everything looks okay